google-cloud-firestore/lib/google/cloud/firestore/document_reference.rb

```diff
@@ -0,0 +1,387 @@
+# Copyright 2017 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/firestore/v1beta1"
+require "google/cloud/firestore/document_snapshot"
+require "google/cloud/firestore/collection_reference"
+
+module Google
+  module Cloud
+    module Firestore
+      ##
+      # # DocumentReference
+      #
+      # A document reference object refers to a document location in a Cloud
+      # Firestore database and can be used to write or read data. A document
+      # resource at the referenced location may or may not exist.
+      #
+      # @example
+      #   require "google/cloud/firestore"
+      #
+      #   firestore = Google::Cloud::Firestore.new
+      #
+      #   # Get a document reference
+      #   nyc_ref = firestore.doc "cities/NYC"
+      #
+      class DocumentReference
+        ##
+        # @private The firestore client object.
+        attr_accessor :client
+
+        ##
+        # The document identifier for the document resource.
+        #
+        # @return [String] document identifier.
+        def document_id
+          path.split("/").last
+        end
+
+        ##
+        # A string representing the path of the document, relative to the
+        # document root of the database.
+        #
+        # @return [String] document path.
+        def document_path
+          path.split("/", 6).last
+        end
+
+        ##
+        # @private A string representing the full path of the document resource.
+        #
+        # @return [String] document resource path.
+        def path
+          @path
+        end
+
+        # @!group Access
+
+        ##
+        # Retrieves a list of collections nested under the document with data.
```

Comment:
I am not sure what you mean by 'with data'. Maybe you can just remove this?

Reply:
Changed to "document snapshot".